### PR TITLE
fix: SentryUser.userId should be nullable

### LINKED ISSUE #2035: `user.id` should be nullable
### Description

The spec defines all fields to be optional: https://develop.sentry.dev/sdk/event-payloads/user/

But we have the `id` to be non-nullable:

https://github.com/getsentry/sentry-cocoa/blob/398f6b24ebd37db618feb74cfb3643e3ff3d003d/Sources/Sentry/Public/SentryUser.h?rgh-link-date=2022-08-02T05%3A25%3A41Z#L12

This means we end up having to pass an empty string as `id` if we want to send a user with only `username`, for example.

### PR DESCRIPTION
**Pull request by kevinrenskers**
## :scroll: Description

`SentryUser.userId` should be nullable to align with the docs / specs.

## :bulb: Motivation and Context

Fixes #2035

## :green_heart: How did you test it?

Unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/Public/SentryUser.h [hunk 1]
```diff
@@ -39,7 +39,6 @@ NS_SWIFT_NAME(User)
 - (instancetype)initWithUserId:(NSString *)userId;
 
 - (instancetype)init;
-+ (instancetype)new NS_UNAVAILABLE;
```
Comment (kevinrenskers) [inline Sources/Sentry/Public/SentryUser.h:42]: I don't see a reason why this was unavailable? The standard init was available too after all 🤔

**Review by brustolin — APPROVED** [review]
LGTM

**Comment by philipphofmann** [comment]
Strictly speaking, this is a breaking change. Are we fine with releasing this in a minor version, @brustolin, @kevinrenskers?

**Comment by kevinrenskers** [comment]
I think the other way around would be a breaking change, if it used to be nullable but now it's not. This is an addition, and existing code should definitely not break. If you gave it an empty string before, that'll still work.

**Comment by philipphofmann** [comment]
Both changes are breaking changes, especially with the nullability features in Swift. If you accessed `userID` before, you could now get a build error that the field is now nullable. 
On the other hand, if you access a nullable field with `!`, and it's not nullable suddenly, you also get a build error.

**Comment by kevinrenskers** [comment]
Hmm right, I see what you mean. But I doubt many people would be reading this value, they have their own user model and only write to this property. I’d be fine with releasing this in a minor version, but we can always revert of course.

**Pull request merged**